runtime/interpreter/interpreter_expression.go

```diff
@@ -1207,8 +1207,6 @@ func (interpreter *Interpreter) VisitCreateExpression(expression *ast.CreateExpr
 func (interpreter *Interpreter) VisitDestroyExpression(expression *ast.DestroyExpression) Value {
 	value := interpreter.evalExpression(expression.Expression)
 
-	interpreter.invalidateResource(value)
```

Comment:
This gets moved to body of the value's `Destroy`, since nested resources are no longer explicitly destroyed with a `destroy` expression.

Reply:
It might be sufficient to keep it here. In `interpreter.invalidateResource`, it only maintains a resource-to-variable mapping. Nested resources are not variables (either they are fields or values at an index), so the nested resources are anyway not tracked by this, if I'm not mistaken. i.e: It only tracks the outer-most resource.

Moving this to `Destroy` of each value might introduce an overhead.

Reply:
Reverted this change.